common/sgx/eeid_verifier.c

```diff
@@ -199,17 +207,29 @@ static oe_result_t _verify_sgx_report(
         }
     }
 
+    sgx_claims_size = sgx_evidence_buffer_size -
+                      (header->report_size + sizeof(oe_report_header_t));
+    *sgx_claims = oe_malloc(sgx_claims_size);
```

Comment:
need to check for allocation failure.   oe_sgx_extract_claims will crash if you pass it a NULL pointer.

In addition, the contract seems odd if we return a buffer to the caller to free even when we return failure from this function (e.g., if the checks fail at 216, 225, etc.), as it's a recipe for memory leak bugs.   I'd recommend changing this function to only return the buffer to the caller when returning OE_OK.

Reply:
Thanks for going through this and sorry for the excessive noise! Looks like a had a bad day when I wrote this, there were quite a few problems with that code.